### PR TITLE
2356 tile group refactor

### PR DESCRIPTION
**Pull request by briaguya**
* Refactor tilegroups for better code reuse
* Implements https://github.com/cityofaustin/techstack/issues/2356

### Home Page
https://deploy-preview-498--janis.netlify.com/
#### Desktop (1080w)
Screen Shot 2019-09-06 at 9 26 00 PM

#### Tablet (880w)
Screen Shot 2019-09-06 at 9 42 21 PM

#### Mobile (375w)
Screen Shot 2019-09-06 at 9 43 05 PM

### Topic Collection Page
https://deploy-preview-498--janis.netlify.com/en/housing-utilities/recycling-trash-and-compost/
#### Desktop (1140w)
Screen Shot 2019-09-06 at 9 28 33 PM

#### Tablet (880w)
Screen Shot 2019-09-06 at 9 37 06 PM

#### Mobile (375w)
Screen Shot 2019-09-06 at 9 37 53 PM

### Topic Page
https://deploy-preview-498--janis.netlify.com/housing-utilities/recycling-trash-and-compost/household-waste/
#### Desktop (1080w)
Screen Shot 2019-09-06 at 9 29 23 PM

#### Tablet (880w)
Screen Shot 2019-09-06 at 9 39 17 PM

#### Mobile (375w)
Screen Shot 2019-09-06 at 9 40 01 PM

### Department Page
https://deploy-preview-498--janis.netlify.com/en/police-oversight/
#### Desktop (1300w)
Screen Shot 2019-09-06 at 9 34 44 PM

#### Tablet (880w)
Screen Shot 2019-09-06 at 9 40 51 PM

#### Mobile (375w)
Screen Shot 2019-09-06 at 9 41 33 PM

### CODE REVIEW

File: src/components/Pages/Departments.js [hunk 1]
```diff
@@ -12,18 +12,16 @@ import TileGroup from 'components/Tiles/TileGroup';
 
 const Departments = ({ departments, intl }) => {
   const departmentLinks = departments.map(d => ({
-    url: `/${intl.locale}/${d.slug}`,
```
Comment (Niicck) [inline src/components/Pages/Departments.js]: Does Tile Group's `export default injectIntl(TileGroup);` add the intl path to the url for us?

File: src/components/Tiles/_Tile.scss [hunk 2]
```diff
@@ -1,44 +1,115 @@
...
+
+  padding: $coa-spacing-small $coa-spacing-small;
+
+  &:hover {
+    border: 1px solid $coa-color-primary;
+  }
+
+  &:hover + .coa-Tile {
+    border-top: none;
+  }
```
Comment (Niicck) [inline src/components/Tiles/_Tile.scss]: ;)

File: src/components/Tiles/TileGroup.js [hunk 3]
```diff
@@ -1,42 +1,46 @@
...
+              ? 'coa-TileGroup__tiles-container--compact'
+              : 'coa-TileGroup__tiles-container'
+          }
         >
-          <Tile url={url} text={text} />
+          {tiles.map(({ type, url, title }, index) => {
+            return (
+              <Tile
+                url={
+                  url.substring(0, 4) === 'http' ? url : `/${intl.locale}${url}`
```
Comment (briaguya) [inline src/components/Tiles/TileGroup.js]: @Niicck the logic to add the language slugs is here now

Comment (Niicck) [reply src/components/Tiles/TileGroup.js]: Awesome. It's a good idea that put that logic in just one place.

Comment (briaguya) [reply src/components/Tiles/TileGroup.js]: @Niicck yeah, it was kinda all over the place, had to do a bit of testing to fix some double language slugs that popped up after consolidating it

Comment (robbobfrh84) [reply src/components/Tiles/TileGroup.js]: 👍This is great. Making a note of this page now.

**Pull request merged**